### PR TITLE
improve code in option parsing routine

### PR DESCRIPTION
**Pull request by ordex**
I hope the code in this patch is similar to what you had in mind when writing the FIXME.

=================================

get rid of FIXME clause and optimize options
parsing code a little bit.

Signed-off-by: Antonio Quartulli <a@unstable.cc>

### CODE REVIEW

File: src/conf.c [hunk 1]
```diff
@@ -61,39 +61,46 @@ int conf_loadfile( conf_t *conf, char *file )
...
 		fgetc( fp );			/* Skip newline */
-		if( strchr( s, '=' ) == NULL )
+		tmp = strchr( s, '=' );
+		if( tmp == NULL )
 			continue;		/* Probably empty? */
+
 		sscanf( s, "%[^= \t]s", key );
+
+		/* start from 1 to skip '=' char */
+		for( i = 1; tmp[i]; i ++ )
```
Comment (ismaell) [inline src/conf.c]: Can we get rid of `i`? i.e. by doing `tmp++`...

Comment (ordex) [reply src/conf.c]: sure, we can. I thought using an index could be cleaner. But it seems you prefer compact code :)

Comment (ismaell) [reply src/conf.c]: I do; plus the compiler doesn't always optimize away the index.

**Comment by ordex** [comment]
Should be quite compressed now :)

File: src/conf.c [hunk 2]
```diff
@@ -77,23 +78,29 @@ int conf_loadfile( conf_t *conf, char *file )
...
 		fgetc( fp );			/* Skip newline */
-		if( strchr( s, '=' ) == NULL )
+		tmp = strchr( s, '=' );
+		if( tmp == NULL )
 			continue;		/* Probably empty? */
+
 		sscanf( s, "%[^= \t]s", key );
+
+		/* skip first char as it is the '=' */
+		while( *( ++ tmp ) )
```
Comment (ismaell) [inline src/conf.c]: I can think of the following improvement to the loop:

	/* Skip the "=" and any spaces following it */ 
	while( isspace( * ++ tmp ) ); /* XXX isspace(0) is false */
	value = tmp;
	while ( *tmp && !isspace( *tmp ) ) tmp++;
	*tmp = 0;

**Comment by ordex** [comment]
I did not know we were at the IOCCC :D 
I applied your suggestion and pushed again.

**Pull request merged**